applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1879,15 +1926,240 @@ void TwoFluidNavierStokes<TElementData>::ComputeSplitting(
+        rInterfaceNormalsNeg,
+        GeometryData::GI_GAUSS_2);
+
+    for (unsigned int gp = 0; gp < rInterfaceNormalsNeg.size(); ++gp){
+        const double normal_norm = norm_2(rInterfaceNormalsNeg[gp]);
+        rInterfaceNormalsNeg[gp] /= normal_norm;
+    }
+
+    // Compute the enrichment shape function values at the interface gauss points using the enrichment interpolation matrices
+    rEnrInterfaceShapeFunctionPos = prod(rInterfaceShapeFunctionNeg, enr_pos_interp);
+    rEnrInterfaceShapeFunctionNeg = prod(rInterfaceShapeFunctionNeg, enr_neg_interp);
+}
+
+template <>
+ModifiedShapeFunctions::UniquePointer TwoFluidNavierStokes< TwoFluidNavierStokesData<2, 3> >::ModifiedShapeFunctionsUtility(
+    const GeometryType::Pointer pGeometry,
+    const Vector& rDistances)
+{
+    auto p_modified_sh_func = Kratos::make_unique<Triangle2D3ModifiedShapeFunctions>(pGeometry, rDistances);
+    return p_modified_sh_func;
+}
+
+template <>
+ModifiedShapeFunctions::UniquePointer TwoFluidNavierStokes< TwoFluidNavierStokesData<3, 4> >::ModifiedShapeFunctionsUtility(
+        const GeometryType::Pointer pGeometry,
+        const Vector& rDistances)
+{
+    auto p_modified_sh_func = Kratos::make_unique<Tetrahedra3D4ModifiedShapeFunctions>(pGeometry, rDistances);
+    return p_modified_sh_func;
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::CalculateCurvature(
+        const Matrix& rInterfaceShapeFunctions,
+        Vector& rInterfaceCurvature)
+{
+    auto geom = this->GetGeometry();
+    const unsigned int n_gpt = rInterfaceShapeFunctions.size1();
+
+    rInterfaceCurvature.resize(n_gpt, false);
+
+    for (unsigned int gpt = 0; gpt < n_gpt; ++gpt){
+        double curvature = 0.0;
+        for (unsigned int i = 0; i < NumNodes; ++i){
+            curvature += rInterfaceShapeFunctions(gpt,i) * geom[i].GetValue(CURVATURE);
+        }
+        rInterfaceCurvature[gpt] = curvature;
+    }
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::SurfaceTension(
+    const double coefficient,
+    const Vector& rCurvature,
+    const Vector& rInterfaceWeights,
+    const Matrix& rInterfaceShapeFunctions,
+    const std::vector<Vector>& rInterfaceNormalsNeg,
+    VectorType& rRHS)
+{
+    VectorType rhs = ZeroVector(NumNodes*NumNodes); //Size is NumNodes x (NumDim+1)
+
+    for (unsigned int intgp = 0; intgp < rInterfaceWeights.size(); ++intgp){
+        for (unsigned int j = 0; j < NumNodes; ++j){
+            for (unsigned int dim = 0; dim < NumNodes-1; ++dim){
+                rhs[ j*(NumNodes) + dim ] -= coefficient*(rInterfaceNormalsNeg[intgp])[dim]
+                    *rCurvature(intgp)*rInterfaceWeights(intgp)*rInterfaceShapeFunctions(intgp,j);
+            }
+        }
+    }
+
+    noalias(rRHS) += rhs;
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::PressureGradientStabilization(
+    TElementData& rData,
```

Comment:
```suggestion
    const TElementData& rData,
```
At first glance, I think this can be `const`.

Reply:
Seems so. I changed it.